File: src/main/java/com/nndmove/app/web/rest/vm/LoginVM.java
package com.nndmove.app.web.rest.vm;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

/**
 * View Model object for storing a user's credentials.
 */
@Getter
@Setter
public class LoginVM {

    @NotNull
    @Size(min = 1, max = 50)
    private String username;

    @NotNull
    @Size(min = 4, max = 100)
    private String password;

    private boolean rememberMe;

    // prettier-ignore
    @Override
    public String toString() {
        return "LoginVM{" +
            "username='" + username + '\'' +
            ", rememberMe=" + rememberMe +
            '}';
    }
}

File: src/main/java/com/nndmove/app/web/rest/vm/LoginVM.kt
package com.nndmove.app.web.rest.vm

import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size

/**
 * View Model object for storing a user's credentials.
 */
class LoginVM {
    @JvmField
    var username: @NotNull @Size(min = 1, max = 50) String? = null

    @JvmField
    var password: @NotNull @Size(min = 4, max = 100) String? = null

    var isRememberMe: Boolean = false

    // prettier-ignore
    override fun toString(): String {
        return "LoginVM{" +
                "username='" + username + '\'' +
                ", rememberMe=" + isRememberMe +
                '}'
    }
}
